src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/UserServiceTest.kt

```diff
@@ -538,24 +548,33 @@ class UserServiceTest {
       assertThat(entity.email).isEqualTo(deliusUser.email)
       assertThat(entity.telephoneNumber).isEqualTo(deliusUser.telephoneNumber)
       assertThat(entity.deliusStaffCode).isEqualTo(deliusUser.staffCode)
+      assertThat(entity.probationRegion.name).isEqualTo(newProbationRegion.name)
 
       verify(exactly = 1) { mockCommunityApiClient.getStaffUserDetails(username) }
       verify(exactly = 1) { mockUserRepository.save(any()) }
     }
 
     @Test
     fun `it stores a null email address if missing from Community API`() {
-      val user = userFactory.produce()
+      val user = userFactory
+        .withUnitTestControlProbationRegion()
+        .produce()
+
       val deliusUser = staffUserDetailsFactory
         .withTelephoneNumber("0123456789")
         .withoutEmail()
+        .withProbationAreaCode(user.probationRegion.deliusCode)
         .produce()
 
       every { mockUserRepository.findByIdOrNull(id) } returns user
       every { mockCommunityApiClient.getStaffUserDetails(username) } returns ClientResult.Success(
         HttpStatus.OK,
         deliusUser,
       )
+      every { mockProbationAreaProbationRegionMappingRepository.findByProbationAreaDeliusCode(user.probationRegion.deliusCode) } returns ProbationAreaProbationRegionMappingEntityFactory()
```

Comment:
I would prefer not to have to add a stub in here because the checking for regions shouldn't impact this behaviour. It seems though the global mocking requires it. Should we move this out of the test and up into the global (or whichever level) context?

Reply:
Yeah, we could hoist it to the top in a `@BeforeEach` function. Something like:

```kotlin
@BeforeEach
fun setup() {
  every { mockProbationAreaProbationRegionMappingRepository.findByProbationAreaDeliusCode(user.probationRegion.deliusCode) } returns ProbationAreaProbationRegionMappingEntityFactory()
}
```